components/server/src/workspace/workspace-service.ts

```diff
@@ -924,13 +924,28 @@ export class WorkspaceService {
         });
     }
 
-    public async resolveBaseImage(ctx: TraceContext, user: User, imageRef: string) {
+    public async validateImageRef(ctx: TraceContext, user: User, imageRef: string) {
         try {
             return await this.workspaceStarter.resolveBaseImage(ctx, user, imageRef);
         } catch (e) {
-            // we could map proper response message according to e.code
-            // see https://github.com/gitpod-io/gitpod/blob/ef95e6f3ca0bf314c40da1b83251423c2208d175/components/image-builder-mk3/pkg/orchestrator/orchestrator_test.go#L178
-            throw ApplicationError.fromGRPCError(e);
+            // see https://github.com/gitpod-io/gitpod/blob/f3e41f8d86234e4101edff2199c54f50f8cbb656/components/image-builder-mk3/pkg/orchestrator/orchestrator.go#L561
+            // TODO(ak) ideally we won't check a message (subject to change)
+            // but ws-manager does not return INTERNAL for invalid image refs provided by a user
+            // otherwise we report it as internal error in observability
+            const code = e["code"];
+            const details = e["details"];
+            if (
+                typeof details === "string" &&
+                (code === grpc.status.INVALID_ARGUMENT || details.includes("cannot resolve image"))
+            ) {
+                let message = details;
+                // strip confusing prefix
+                if (details.startsWith("cannt resolve base image ref: ")) {
+                    message = details.substring("cannt resolve base image ref: ".length);
```

Comment:
Respond code with be something else, see [test case which use it](https://github.com/gitpod-io/gitpod/blob/ef95e6f3ca0bf314c40da1b83251423c2208d175/components/image-builder-mk3/pkg/orchestrator/orchestrator_test.go#L167-L181)

nit-fixing since it doesn't break UX

<img width="769" alt="image" src="https://github.com/gitpod-io/gitpod/assets/20944364/eb5edb64-674b-47f2-8dee-57607b895b92">

Reply:
I'm not sure what I should change?

Reply:
Based on the code you added, you want to remove the prefix `cannot resolve image` but it will not match since the respond rpc code is not `grpc.status.INVALID_ARGUMENT`

So this check is useless ()

Not a blocker

Reply:
I think it is covered because it is `||` not `&&`, but I agree though that explicit check for NOT_FOUND and UNAUTHORIZED makes sense here too. 

We need generally align how we propagate errors thorugh the system and we don't break contracts.